modules/core/src/lifecycle/component-state.js

```diff
@@ -34,6 +34,16 @@ export default class ComponentState {
     this.oldAsyncProps = null; // Last props before update, with async values copied.
   }
 
+  finalize() {
+    for (const propName in this.asyncProps) {
+      const asyncProp = this.asyncProps[propName];
+      if (asyncProp.type && asyncProp.type.transform) {
+        // Release any resources created by transforms
+        asyncProp.type.transform(null, asyncProp.resolvedValue, asyncProp.type, this.component);
```

Comment:
Nit: Not a big deal, but I think it feels a bit weird to call a method named `transform` here (instead of e.g. `delete`). 

I would not have guessed it, so we may want to mention in the docs that this will be called at the end with a `null`.

Also this means that every transform function needs to check for `null` (or maybe they already did?).

Reply:
I originally had two callbacks `transform` and `release`. Though having `transform` receiving both the old and the new values allows us more flexibility in managing resources (e.g. updating a buffer instead of deleting and creating a new one). Then `release` started to seem redundant.

Any transofrm function will have to check for `null` regardless, in case loader fails.

Reply:
Yes sure.

> Any transofrm function will have to check for null regardless, in case loader fails.

I assume we could have added a check before calling `transform`...